add missing check of MFT_RECORD fields, and several debugs

jaysim opened the pull request:
Add missing check of MFT_RECORD and when clearing MFT bitmap, also make mft record unused.

jaysim committed:
ntfsck: add missing check logic of MFT_RECORD fields

check sequence number of MFT_RECORD and indexed_file of index entry.
check link count is zero.
check base_mft_record is not zero.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

jaysim committed:
ntfsck: add clearing mft bit and making mft unused

When clearing mft bitmap, also make mft record unused
(clear IN_USED bit of mft record)

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

jaysim committed:
ntfsck: add file flags checking same as kernel

when inode open in kernel, check inode's flags(MFT flags),
so fsck also add checking flags.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

jaysim committed:
ntfsck: add check return value of ntfs_attrs_walk()

ntfs_attrs_walk() return ENOENT, when walk to end of inode attributes successfully.
If it encounter error, it return EIO.
at least, fsck should be distingushed it at first walk.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

namjaejeon commented on src/ntfsck.c:
```diff
@@ -2629,6 +2682,10 @@ static int ntfsck_check_system_inode(ntfs_inode *ni, INDEX_ENTRY *ie,
 {
 	int ret;
 
+	if (ntfsck_check_inode_fields(ictx->ni, ni, ie)) {
```
braces {} are not necessary for single statement blocks

namjaejeon commented on src/ntfsck.c:
```diff
@@ -2586,6 +2635,10 @@ static int ntfsck_check_inode(ntfs_inode *ni, INDEX_ENTRY *ie,
 	int32_t flags;
 	int ret;
 
+	if (ntfsck_check_inode_fields(ictx->ni, ni, ie)) {
```
braces {} are not necessary for single statement blocks

jaysim committed:
ntfsprogs-plus v0.9.2

Change Note:
handle MFT bitmap free but MFT record is in-use
recover $MFT and $MFTMirr record are different
add missing check of MFT Record fields
add file flags check same as kernel does
check return value of ntfs_attrs_walk()

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>
